File: Rust/5 Kyu/maximum_subarray_sum.rs
use std::cmp;

fn max_sequence(seq: &[i32]) -> i32 {
    
    let tam = seq.len();
    
    let mut soma = 0;
    
    let mut maior = 0;
    
    for x in 0..tam{
        
        soma = cmp::max(seq[x] , soma+seq[x]);
        maior = cmp::max(maior , soma);
        
    }
    
    maior
    
}

File: C/5 Kyu/maximum_subarray_sum.c


int max(a,b){
  if( a>=b){
    return a;
  }else{
    return b;
  }
}


int maxSequence(const int* array, size_t n) {
  
    size_t i;
    
    int soma = 0;
    
    int maior = 0;
  
    for(i = 0; i<n;i++){
      
      soma = max(array[i],soma+array[i]);
      maior = max(maior,soma);
      
      
    }

  
  return maior;
}